src/traces/carpet/map_1d_array.js

```diff
@@ -16,7 +18,7 @@
 module.exports = function mapArray(out, data, func) {
     var i;
 
-    if(!Array.isArray(out)) {
+    if(!isArrayOrTypedArray(out)) {
         // If not an array, make it an array:
         out = [];
     } else if(out.length > data.length) {
```

Comment:
⚠️ this block has a `.slice` ⚠️

Reply:
That's ok though:

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/TypedArray/slice

Reply:
Ah great. So this is fine, but I guess 🐎 at some point we could make a helper that uses `slice` on regular arrays and `subarray` on typed arrays, for cases like this where we don't need a copy.